src/core/handlers/RequestHandler.ts

```diff
@@ -314,37 +318,38 @@ export abstract class RequestHandler<
     resolver: ResponseResolver<ResolverExtras>,
   ): ResponseResolver<ResolverExtras> {
     return async (info): Promise<ResponseResolverReturnType<any>> => {
-      const result = this.resolverGenerator || (await resolver(info))
-
-      if (isIterable<AsyncResponseResolverReturnType<any>>(result)) {
-        // Opt-out from marking this handler as used.
-        this.isUsed = false
-
-        if (!this.resolverGenerator) {
-          this.resolverGenerator = result
+      if (!this.resolverGenerator) {
+        const result = await resolver(info)
+        if (!isIterable(result)) {
+          return result
         }
+        this.resolverGenerator =
+          Symbol.iterator in result
```

Comment:
`Reflect.has` (used in `isIterable`) is the same operation as `in`, except that TS can acutally narrow using `in`.